tokio/src/runtime/queue.rs

```diff
@@ -49,24 +35,11 @@ pub(super) struct Inner<T: 'static> {
     tail: AtomicU16,
 
     /// Elements
-    buffer: Box<[UnsafeCell<MaybeUninit<task::Notified<T>>>]>,
-}
-
-struct Pointers {
-    /// True if the queue is closed
-    is_closed: bool,
-
-    /// Linked-list head
-    head: Option<NonNull<task::Header>>,
-
-    /// Linked-list tail
-    tail: Option<NonNull<task::Header>>,
+    buffer: Box<[UnsafeCell<MaybeUninit<task::Notified<T>>>; LOCAL_QUEUE_CAPACITY]>,
```

Comment:
i guess the other question re https://github.com/tokio-rs/tokio/pull/3939/files#r667212885 is, does this even need to be boxed anymore, if it has a fixed size? should we save another layer of indirection by removing the `Box`, or is there a reason for this to live in a separate allocation beyond sizedness?

Reply:
It doesn't need to, no. If I am to come up with a reason, then some allocators probably dislike allocations that are slightly larger than 1024 bytes. I don't know what effect this has in practice. I am open to removing the box.

Reply:
I guess another reason is that long fixed size arrays are really awkward to construct for non-copy generic types.